### ISSUE TITLE
Unrecognised token is not an error in many cases

### ISSUE THREAD

**Issue by seanyoung** [bengtmartensson/IrpTransmogrifier#238]
The error gets reported to stdout, but no error is reported to the caller when `org.harctoolbox.irp.Protocol` is instantiated via java. This is visible via the cli interface:
```bash
$ ./irptransmogrifier.sh --irp '{}<-1>(1
,X3="8,-7)' render -r 
line 2:4 token recognition error at: '"'
Freq=38000Hz[+1,-7][][]
```
This means that if I use IrpTransmogrifier via the java interface, I have no idea whether the IRP was correct or not. 
There are many examples of this, each time an error is reported but the IRP is rendered anyway.
```bash
$ ./irptransmogrifier.sh --irp '{}<1>($1,-100)' render -r 
line 1:6 token recognition error at: '$'
Freq=38000Hz[+1,-100][][]
```

**Comment by bengtmartensson** [comment]
Sean, although this did not make it to the just released 1.2.12, I will address it shortly. It is simply the lexer that does not barf when it sees a token it does not know.

**Comment by bengtmartensson** [comment]
Hope you find the fix OK. From Java, a (non-checked) ParseCancellationException is thrown.
